src/wwwroot/js/genpage/main.js

```diff
@@ -16,6 +16,101 @@ let allModels = [];
 
 let coreModelMap = {};
 
+if (typeof window.modelArchitectureMap === 'undefined') {
+    window.modelArchitectureMap = {};
+}
+let modelArchitectureMap = window.modelArchitectureMap;
+
+function loadModelArchitectureMap() {
+    for (let key in modelArchitectureMap) {
+        delete modelArchitectureMap[key];
+    }
+    
+    if (!allModels || allModels.length === 0) {
+        return;
+    }
+    
+    let processedCount = 0;
+    const totalModels = allModels.length;
+    
+    for (let modelName of allModels) {
+        genericRequest('DescribeModel', {'modelName': modelName, 'subtype': 'Stable-Diffusion'}, data => {
+            processedCount++;
+            
+            if (!data.model) {
+                modelArchitectureMap[modelName] = 'Unknown';
+                return;
+            }
+            
+            const model = data.model;
+            
+            if (model.architecture_group) {
+                modelArchitectureMap[modelName] = model.architecture_group;
+            } 
+            else if (model.architecture) {
+                const grouped = groupArchitecture(model.architecture);
+                modelArchitectureMap[modelName] = grouped;
+            } 
+            else {
+                modelArchitectureMap[modelName] = 'Unknown';
+            }
+            
+            if (processedCount === totalModels) {
+                if (typeof imageHistoryBrowser !== 'undefined' && imageHistoryBrowser) {
+                    imageHistoryBrowser.update();
+                    updateArchitectureDropdown();
+                }
+            }
+        }, undefined, error => {
+            processedCount++;
+            modelArchitectureMap[modelName] = 'Unknown';
+            
+            if (processedCount === totalModels) {
+                if (typeof imageHistoryBrowser !== 'undefined' && imageHistoryBrowser) {
+                    imageHistoryBrowser.update();
+                    updateArchitectureDropdown();
+                }
+            }
+        });
+    }
+}
+
+
+function groupArchitecture(arch) {
+    if (!arch) return null;
+    
+    const archLower = arch.toLowerCase();
+    
+    if (archLower.startsWith('flux-1') || arch.startsWith('Flux.1')) return 'Flux.1';
+    
+    if (archLower.startsWith('stable-diffusion-xl')) return 'SDXL';
```

Comment:
This whole function is a weird hack. T2IModelClassSorter is where data like this would be defined